tests/jq.test

```diff
@@ -604,6 +604,10 @@ def f(a;b;c;d;e;f): [a+1,b,c,d,e,f]; f(.[0];.[1];.[0];.[0];.[0];.[0])
 [1,2]
 [2,2,1,1,1,1]
 
+def f: 1; def g: f, def f: 2; def g: 3; f, def f: g; f, g; def f: 4; [f, def f: g; def g: 5; f, g]+[f,g]
```

Comment:
Oof, this one is hard to parse.  For anyone following along:

```
def f: 1;
def g: f,
       def f: 2;
       def g: 3;
       f,
       def f: g;
       f, g;
def f: 4;
[f, def f: g; def g: 5; f, g]+[f,g]
```
The first `f` in `[f, ...]` will be `4`, the second will be `g`, which will b `1, 2, 3, 3`, then `g` will be `5`.  The we add `[f,g]` to that, which will be the first `f` and `g`, so, again `[1,2,3,3]`.